custom_model_runner/datarobot_drum/drum/gpu_predictors/nemo_predictor.py

```diff
@@ -0,0 +1,239 @@
+"""
+Copyright 2024 DataRobot, Inc. and its affiliates.
+All rights reserved.
+This is proprietary source code of DataRobot, Inc. and its affiliates.
+Released under the terms of DataRobot Tool and Utility Agreement.
+"""
+import csv
+import io
+import logging
+import os
+import signal
+import subprocess
+import atexit
+
+import numpy as np
+from openai import OpenAI
+from threading import Thread
+from datarobot_drum import RuntimeParameters
+from datarobot_drum.drum.adapters.model_adapters.python_model_adapter import RawPredictResponse
+from datarobot_drum.drum.common import SupportedPayloadFormats
+from datarobot_drum.drum.enum import (
+    LOGGER_NAME_PREFIX,
+    PayloadFormat,
+    StructuredDtoKeys,
+)
+from datarobot_drum.drum.exceptions import DrumCommonException
+from datarobot_drum.drum.language_predictors.base_language_predictor import BaseLanguagePredictor
+from datarobot_drum.resource.drum_server_utils import wait_for_server, DrumServerProcess
+
+RUNNING_LANG_MSG = "Running environment: Nemo Inference Microservices."
+DEFAULT_MODEL_NAME = "generic_llm"
+
+
+class ChatRoles:
+    SYSTEM = "system"
+    ASSISTANT = "assistant"
+    USER = "user"
+
+
+class NemoPredictor(BaseLanguagePredictor):
+    def __init__(self):
+        super(NemoPredictor, self).__init__()
+        self.logger = logging.getLogger(LOGGER_NAME_PREFIX + "." + __name__)
+        self.datarobot_venv_path = os.environ.get("DATAROBOT_VENV_PATH", "/home/nemo/dr")
```

Comment:
Aren't we running inside of the venv already? e.g. didn't we install and start drum up inside of the venv so why do we need to know it's path? We can deduce that from the location of the python binary we are running as right?

Reply:
updated to os.environ.get('VIRTUAL_ENV')